dashboard/app/models/programming_method.rb

```diff
@@ -0,0 +1,23 @@
+# == Schema Information
+#
+# Table name: programming_methods
+#
+#  id                   :bigint           not null, primary key
+#  programming_class_id :integer
+#  key                  :string(255)
```

Comment:
For my understanding, what do keys look like for methods? If I'm reading the spec correctly, we don't use them in URLs -- are they established by the documentation writer?

Reply:
I was planning on using them for the div ids so that the methods can be linked to directly, as well as for the setting function overloading like we were talking about yesterday. For how they'll look, my plan was to generate them like `<method name>-<param 1>-<param 2>-...` but we could also let curriculum writers set them if we wanted to